packages/instrumentation-amqplib/test/amqplib-promise.spec.ts

```diff
@@ -0,0 +1,514 @@
+import 'mocha';
+import expect from 'expect';
+import sinon from 'sinon';
+import lodash from 'lodash';
+import { InMemorySpanExporter, SimpleSpanProcessor } from '@opentelemetry/tracing';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import { HttpTraceContext } from '@opentelemetry/core';
+import { AmqplibInstrumentation, EndOperation, PublishParams } from '../src';
+
+const instrumentation = new AmqplibInstrumentation();
+instrumentation.enable();
+instrumentation.disable();
+
+import amqp from 'amqplib';
+import { GeneralAttribute, MessagingAttribute } from '@opentelemetry/semantic-conventions';
+import { propagation, Span, SpanKind, SpanStatusCode } from '@opentelemetry/api';
+import { asyncConsume } from './utils';
+
+const msgPayload = 'payload from test';
+const queueName = 'queue-name-from-unittest';
+
+describe('amqplib instrumentation promise model', function () {
```

Comment:
I suggest to add another test that confirms instrumentation.disable() works well

1. create publish.
2. disable instrumentation
3. create publish
4. confirm only 1 span created